wilds/common/grouper.py

```diff
@@ -87,10 +87,11 @@ def __init__(self, dataset, groupby_fields):
             # Note that this might result in some empty groups.
             self.groupby_field_indices = [i for (i, field) in enumerate(dataset.metadata_fields) if field in groupby_fields]
             if len(self.groupby_field_indices) != len(self.groupby_fields):
-                raise ValueError('at least one group field not found in dataset.metadata_fields')
+                raise ValueError('At least one group field not found in dataset.metadata_fields')
             grouped_metadata = dataset.metadata_array[:, self.groupby_field_indices]
             if not isinstance(grouped_metadata, torch.LongTensor):
-                warnings.warn(f'CombinatorialGrouper: converting metadata with fields [{", ".join(groupby_fields)}] into long')
+                if not torch.all(grouped_metadata == grouped_metadata.long()):

```

Comment:
could put `grouped_metadata.long()` into a variable since you are going to compute it anyway